### PR TITLE
Created post comment endpoint. Return later.

### PR DESCRIPTION
**Pull request by WiccanWolf**
Unable to debug or complete 201 status. Returning later.

### CODE REVIEW

File: express-app/app.models.js [hunk 1]
```diff
@@ -61,3 +62,29 @@ exports.fetchComments = (article_id) => {
...
+    return Promise.reject({ status: 400, msg: 'Bad Request' });
+  }
+  if (body.length < 1) {
+    return Promise.reject({ status: 400, msg: 'Bad Request' });
+  }
+  const query = format(
+    'INSERT INTO comments (article_id, author, body) VALUES (%L, %L, %L) RETURNING *;',
+    article_id,
+    author,
+    body
```
Comment (nc-mezz-davies) [inline express-app/app.models.js:77]: This syntax is not quite correct for the SQL query. No need to use format. You need to use the $ syntax for the values in the string and to pass an array with those variables within it as the second argument to `db.query()`. This should hopefully pass your 201 test.

File: express-app/app.models.js [hunk 2]
```diff
@@ -61,3 +62,29 @@ exports.fetchComments = (article_id) => {
...
+  );
+  return db.query(query).then(({ rows }) => {
+    const newComment = rows[0];
+    return {
```
Comment (nc-mezz-davies) [inline express-app/app.models.js:81]: You can just return the comment that is retuned from the successful insertion. Votes will be 0 - look at the seed function to confirm this.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -195,3 +195,67 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+});
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:261]: We should also test that the if author of the comment _isn't found_ in the users table we get a suitable error code back. PSQL will be your friend for this circumstance - console the error in the catch to see how to proceed when running the test

File: __tests__/app.test.js [hunk 4]
```diff
@@ -195,3 +195,67 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+
+  test('400: responds with "Bad Request" if body is too short (less than 1 character)', () => {
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:248]: This will most likely be handled on the front end via form validation (not something you will have seen yet, of course) but good that you have thought of this edge case.

File: __tests__/app.test.js (hunk 3 above)
Comment (nc-mezz-davies) [reply __tests__/app.test.js:261]: We also need to handle if the id is valid but nonexistent eg `/api/articles/99999/comments` as well as if the id is invalid eg `/api/not_an_id/comments`
